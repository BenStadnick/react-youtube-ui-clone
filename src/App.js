import React, { useState, useEffect } from 'react';
import youtube from './api/youtube';
import { Container, Row, Col } from 'react-bootstrap';
import { SearchBar, VideoList, VideoPlayer } from './components';

import 'bootstrap/dist/css/bootstrap.min.css';

const App = () => {
  const [videos, setVideos] = useState([]);
  const [selectedVideo, setSelectedVideo] = useState(null);
  
  const defaultSearchTerm = 'Python';
  useEffect(() =>{
    handleSubmit(defaultSearchTerm);
  }, []);

  const handleSubmit = async (searchTerm) => {
    const response = await youtube.get('search', {
      params: {
        part: 'snippet',
        type: 'video',
        videoEmbeddable: 'true',
        maxResults: 5,
        key: `${process.env.REACT_APP_YOUTUBE_API_KEY}`,
        q: searchTerm,
      }
    });

    setSelectedVideo(response.data.items[0]);
    setVideos(response.data.items);
  }

  const onVideoSelect = (videoId) => {
    setSelectedVideo(videoId);
  }

  return (
    <Container>
      <Row>
        <Col md={12}>
          <SearchBar onFormSubmit={handleSubmit} defaultSearchTerm={defaultSearchTerm} />
        </Col>
      </Row>
      <Row>
        <Col md={8}>
          <VideoPlayer video={selectedVideo} />
        </Col>
        <Col md={4}>
          <VideoList videos={videos} onVideoSelect={onVideoSelect} />
        </Col>
      </Row>   
    </Container>
  );
}

export default App;