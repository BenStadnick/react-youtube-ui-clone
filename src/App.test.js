import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import App from './App';

describe('Top level \'app\' component', () => {
  it('did shallow render', () => {
    const wrapper = shallow(<App debug />);
    expect(wrapper).toMatchSnapshot();

  });
});