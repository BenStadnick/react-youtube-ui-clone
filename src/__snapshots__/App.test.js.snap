// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top level 'app' component did shallow render 1`] = `
<Container
  fluid={false}
>
  <ForwardRef
    noGutters={false}
  >
    <Col
      md={12}
    >
      <SearchBar
        defaultSearchTerm="Python"
        onFormSubmit={[Function]}
      />
    </Col>
  </ForwardRef>
  <ForwardRef
    noGutters={false}
  >
    <Col
      md={8}
    >
      <VideoPlayer
        video={null}
      />
    </Col>
    <Col
      md={4}
    >
      <VideoList
        onVideoSelect={[Function]}
        videos={Array []}
      />
    </Col>
  </ForwardRef>
</Container>
`;
